plugins/woocommerce-blocks/assets/js/blocks-registry/product-collection/register-product-collection.tsx

```diff
@@ -64,9 +70,27 @@ const registerProductCollection = ( {
 		);
 	}
 
-	registerBlockVariation( blockJson.name, {
+	const isActive = (
+		blockAttrs: BlockAttributes,
+		variationAttributes: BlockAttributes
+	) => {
+		return blockAttrs.collection === variationAttributes.collection;
+	};
+
+	registerBlockVariation( BLOCK_NAME, {
 		...blockVariationArgs,
+		attributes: {
+			...DEFAULT_ATTRIBUTES,
+			...blockVariationArgs.attributes,
+			query: {
+				...DEFAULT_ATTRIBUTES.query,
+				...blockVariationArgs.attributes?.query,
+			},
+			collection: blockVariationArgs.name,
+			inherit: blockVariationArgs.attributes?.inherit || false,
+		},
+		innerBlocks: blockVariationArgs.innerBlocks || INNER_BLOCKS_TEMPLATE,
```

Comment:
As we discussed elsewhere, I think we should expose `hideControls` more explicitly than through `attributes`. I'm also thinking the same about `query`, it's important bit to configure and maybe that should not be nested in `attributes` on public API level.

It's just a first iteration of API but in general I'm thinking that instead of exposing `attributes` object, we should explicitly expose specific `attributes` in the API.

So basically, I'd personally see something like this:

```
wc.wcBlocksRegistry.__experimentalRegisterProductCollection({
  name: "woocommerce/product-collection/my-custom-collection-with-preview",
  ...
  hideControls: [ 'on-sale', 'featured' ],
  query: {
    woocommerceOnSale: true,
  },
  // There should be validation, this key is not part of public API and hence will be ignored.
  convertedFromProducts: true 
});
```

instead of:

```
wc.wcBlocksRegistry.__experimentalRegisterProductCollection({
  name: "woocommerce/product-collection/my-custom-collection-with-preview",
  ...
  attributes: {
      hideControls: [ 'on-sale', 'featured' ],
      query: {
        woocommerceOnSale: true,
      },
      // User can pass any attribute even if not intended to be used.
      convertedFromProducts: true
  }
});
```

At first, I'd expose:
- `query`
- `displayLayout`
- `collection`
- `hideControls`
- `queryContextIncludes`

But I'm open to discuss it if you think otherwise.

Reply:
I'm not entirely convinced about condensing the attributes into specific arguments, but the inclusion of private "utility" attributes like `convertedFromProducts` makes me believe that Karol might be onto something. Additionally, given that these collections will be part of a dedicated experience - the collection chooser - it makes even more sense to keep this validated.

A few more critical notes while working with attributes:
- The `queryContextIncludes` contains information that should not be overridden.
- The `collection` attribute depends on the selected variation; thus, it should not be edited.
- The `convertedFromProducts` should not be overridden; possibly others that exist now or in the future (like the temp one handling the preview.)

Now, onto the question of whether to introduce curated arguments. For now, I'll say no. The reason is that I want to take a step back and consider what makes common sense here. Describing these settings as "attributes" helps foster a better understanding of how things work. Eventually, this approach may lead to the point where writing product collections with freestyle writing within templates feels natural.

Instead, we can validate the attributes accepted in this helper. Consider the following spec:

Top-level arguments:
- `name`*
- `title`*
- `description`
- `category`
- `keywords`
- `icon`
- `innerBlocks`
- `example` 
- `attributes` (optional, validated)

Accepted attributes:
- `query`
- `displayLayout`
- `hideControls`
- (arguably, for now) `queryContextIncludes`: _Only additions, not replacements._

Restricted arguments:
- `scope` 
- `isDefault`
- `isActive`

_*: Required items_

This is a curated version of the familiar arguments of the _Variations API_, providing space for reconsideration without introducing a completely new schema at this time.

What do you all think?

Reply:
I think I like that... it's a nice middle ground between the custom API I proposed and regular variation API. Assuming users are familiar with the concept of blocks, block attributes, and variations, keeping it as you suggested @xristos3490 feels more natural and like a natural extension of existing APIs.

I like what you suggested!

Reply:
I also like what @xristos3490 suggested 💜

> Accepted attributes:> 
> - `query`

In query, We can also restrict properties like:
- `postType`: Because we always want it to be "product".
- `isProductCollectionBlock`: Because we always want it to be true.

> - (arguably, for now) `queryContextIncludes`: _Only additions, not replacements._

JFYI We will always have the default attributes in `queryContextIncludes` as you can see [here](https://github.com/woocommerce/woocommerce/blob/ec145fc43c82c817b6b93caaff305207553ab578/plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx#L177-L181). For now, we will always have "collection" in `queryContextIncludes`.

---

Adding validation for arguments is in scope of #47963. So I will not add validation for arguments in this PR. Except adding validations, I will make the other changes in this PR. Does that sound good?

Reply:
> scope (?)

For now, it's being used by Core collections because we don't want to show our core collections inside the inserter. Maybe we can allow it for now.
- Maybe 3PDs can use it to hide their collections from the inserter if they wanna use it in themes only but shouldn't be shown in inserter. OR maybe they can use it for some other purpose. What do you think?

Reply:
> For now, we will always have "collection" in queryContextIncludes.

That's good to know! 👏 

> In query, We can also restrict properties like:
postType: Because we always want it to be "product".
isProductCollectionBlock: Because we always want it to be true.

Yes! 100% agree - nice catch! 

> Maybe 3PDs can use it to hide their collections from the inserter if they wanna use it in themes only but shouldn't be shown in inserter. OR maybe they can use it for some other purpose. What do you think?

Don't have a strong preference, tbh. No problem if you feel like exposing it - we can reiterate if required :)

Reply:
Made the changes in [ab9690f](https://github.com/woocommerce/woocommerce/pull/48141/commits/ab9690f171e263cdc927445aaedb31a919d114a9)